modules/apigee_edge_apidocs/src/Form/ApiDocSettingsForm.php

```diff
@@ -41,13 +42,55 @@ public function getFormId() {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
+    $entity_type = \Drupal::entityTypeManager()
+      ->getStorage('apidoc')
+      ->getEntityType();
+    $config = $this->configFactory()->getEditable('apigee_edge_apidocs.settings');
+
+    $options = $form_state->getValue('options');
+    $config->set('default_revision', (bool) $options['new_revision'])->save();
+
+    $args = [
+      '@type' => $entity_type->getLabel(),
+    ];
+    drupal_set_message($this->t('@type settings have been updated.', $args));
   }
 
   /**
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
     $form['apidoc_settings']['#markup'] = $this->t('Settings for API Docs. Manage field settings using the tabs above.');
+
+    $form['additional_settings'] = [
+      '#type' => 'vertical_tabs',
+    ];
+
+    $form['workflow'] = [
+      '#type' => 'details',
+      '#title' => t('Publishing options'),
+      '#group' => 'additional_settings',
+    ];
+    $workflow_options = [
+      'new_revision' => ApiDoc::shouldCreateNewRevision(),
```

Comment:
I would avoid calling static methods (See my comment above) and hardcoding our ApiDoc class to the code. Default entity classes can be swapped by custom modules.
The current entity type object that belongs to the API documentation entity type should be retrieved from the entity type manager and the `shouldCreateNewRevision()` method should be called on that object.

Reply:
Fixed.